### PR TITLE
[AAE-17458] Move default apps away from repository

### PR DESCRIPTION
**Pull request by MichalFidor**
**Please check if the PR fulfills these requirements**

> - [x] The commit message follows our [guidelines](https://github.com/Alfresco/alfresco-ng2-components/wiki/Commit-format)
> - [ ] Tests for the changes have been added (for bug fixes / features)
> - [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")

> - [ ] Bugfix
> - [ ] Feature
> - [ ] Code style update (formatting, local variables)
> - [ ] Refactoring (no functional changes, no api changes)
> - [ ] Build related changes
> - [ ] Documentation
> - [x] Other... Please describe:

**What is the current behaviour?** (You can also link to an open issue here)
Currently we are relying on the default applications which are in this repository. The problem is that it's public and we have them zipped which makes harder to track what change has been made in the particular PR.

**What is the new behaviour?**
This PR contains:
- Removing the apps from this repository,
- Added the logic of downloading the repo as zip and then unzipping it and zip again the particular app - seems that it could be simplified, but - the other repo is private, so we need to use a token. Because of that, I need to use a GH API which allows only to download whole repo as zip. There is no possibility to download one particular thing,
- Introduced a new logic to the script `init-aee-env`. Let me know, if I need to update also the two other - **I don't think I need**
- I've removed unneeded information from `resource.ts` file as it's not needed anymore - most probably this file should be also moved to the private repo 🤔 - **it's needed for hxp-frontend apps, bringing it back**

**Does this PR introduce a breaking change?** (check one with "x")
I think that...
> - [x] Yes
> - [ ] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:
- We need to generate a new token, that will have a permissions to private repositories inside of Hyland org
- Other things that I've missed and hopefully you will catch during the PR review

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -205,8 +206,7 @@
     "tsconfig-paths": "^4.1.1",
     "typescript": "4.7.4",
     "webdriver-manager": "12.1.9",
-    "webpack": "^5.88.2",
-    "webpack-cli": "^5.1.2"
```
Comment (acichon-h) [inline package.json:209]: Is this change intentional?

File: lib/cli/scripts/init-aae-env.ts [hunk 2]
```diff
@@ -94,7 +98,7 @@ async function healthCheck(nameService: string) {
  */
 async function getApplications(): Promise<{ list: { entries: any[] } }> {
     const url = `${args.host}/deployment-service/v1/applications`;
-
+    logger.info(url)
```
Comment (acichon-h) [inline lib/cli/scripts/init-aae-env.ts:101]: I believe it would be good to log some description besides url for context

File: lib/cli/scripts/github-folder-downloader.ts [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    private token: string;
+    private repo: string;
+    private subFolderName: string;
+
+    constructor(token: string, repo: string) {
+        this.token = token;
+        this.repo = repo;
+    }
+
+    async downloadRepoAndUnpackRepository(owner: string, branch = 'main'): Promise<void> {
```
Comment (acichon-h) [inline lib/cli/scripts/github-folder-downloader.ts:16]: This doesn't sound right 😅 
```suggestion
    async downloadAndUnpackRepository(owner: string, branch = 'main'): Promise<void> {
```

File: lib/cli/scripts/github-folder-downloader.ts [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+            Accept: 'application/vnd.github+json',
+            'X-GitHub-Api-Version': '2022-11-28'
+        };
+
+        try {
+            const response: AxiosResponse = await axios.get(url, {
+                responseType: 'arraybuffer',
+                headers,
+            });
+            const repoZipped = new AdmZip(response.data);
```
Comment (acichon-h) [inline lib/cli/scripts/github-folder-downloader.ts:29]:
```suggestion
            const zippedRepo = new AdmZip(response.data);
```

File: lib/cli/scripts/github-folder-downloader.ts [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+            this.subFolderName = (repoZipped.getEntries().find(entry => entry.entryName.includes(owner))).entryName;
```
Comment (acichon-h) [inline lib/cli/scripts/github-folder-downloader.ts:30]: `Subfolder` is a one word
```suggestion
            this.subfolderName = (zippedRepo.getEntries().find(entry => entry.entryName.includes(owner))).entryName;
```

File: lib/cli/scripts/github-folder-downloader.ts [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+
+            new AdmZip(response.data).extractAllTo(this.repo, /** overwrite **/ true);
```
Comment (acichon-h) [inline lib/cli/scripts/github-folder-downloader.ts:32]: I'm not sure if this comment should stay here
```suggestion
            new AdmZip(response.data).extractAllTo(this.repo, true);
```

File: lib/cli/scripts/github-folder-downloader.ts [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+    async zipNeededApplication(appName: string): Promise<string> {
+        const pathToZip = path.resolve(`${process.cwd()}/${this.repo}`, `${appName}.zip`);
+        try {
+            const zip = new AdmZip();
+            zip.addLocalFolder(`${process.cwd()}/${this.repo}/${this.subFolderName}${appName}`);
+            await zip.writeZipPromise(pathToZip);
+        } catch (e) {
+            console.log(`Error during zipping app ${e}`)
+        }
+        return pathToZip;
```
Comment (acichon-h) [inline lib/cli/scripts/github-folder-downloader.ts:48]: Does it make any sense to return path to zip if zipping failed?

Comment (MichalFidor) [reply lib/cli/scripts/github-folder-downloader.ts:48]: Basically it should throw an error, I've missed that!

File: package.json (hunk 1 above)
Comment (MichalFidor) [reply package.json:209]: Nope, I've removed it by mistake

File: lib/cli/scripts/init-aae-env.ts [hunk 8]
```diff
@@ -117,12 +119,10 @@ async function getApplications(): Promise<{ list: { entries: any[] } }> {
             contentTypes,
             accepts
         );
-        result.on('error', (error) => {
-            logger.error(`Get application by status ${error} `);
-        });
+        // logger.info(result)
```
Comment (DenysVuika) [inline lib/cli/scripts/init-aae-env.ts:122]: commented out code

File: lib/cli/scripts/github-folder-downloader.ts [hunk 9]
```diff
@@ -0,0 +1,50 @@
...
+                owner,
+                ref: branch,
+             })
+
+            const parsedResponse = Buffer.from(<ArrayBuffer>response.data);
+            const zippedRepo = new AdmZip(parsedResponse);
+            this.subfolderName = (zippedRepo.getEntries().find(entry => entry.entryName.includes(owner))).entryName;
+
+            zippedRepo.extractAllTo(this.repo, true);
+
```
Comment (acichon-h) [inline lib/cli/scripts/github-folder-downloader.ts:32]: Please remove redundant empty lines

File: lib/cli/scripts/github-folder-downloader.ts [hunk 10]
```diff
@@ -0,0 +1,50 @@
...
+        } catch (error) {
+            throw new Error(`Failed to download repository: ${error}`);
+        }
+    }
+
+    async zipNeededApplication(appName: string): Promise<string> {
```
Comment (acichon-h) [inline lib/cli/scripts/github-folder-downloader.ts:38]:
```suggestion
    async zipApplication(appName: string): Promise<string> {
```

File: lib/cli/scripts/resources.ts [hunk 11]
```diff
@@ -19,196 +19,192 @@
...
-        },
-        security: [
-            { role: 'APPLICATION_MANAGER', groups: [], users: ['manageruser'] },
-            { role: 'ACTIVITI_ADMIN', groups: [], users: ['superadminuser', 'processadminuser'] },
-            { role: 'ACTIVITI_USER', groups: ['hr', 'sales', 'testgroup'], users: ['hruser'] }
-        ],
-        infrastructure: { connectors: { restconnector: {} }, bridges: {} },
-        enableLocalDevelopment: true
-    },
+    // SUB_PROCESS_APP: {
```
Comment (acichon-h) [inline lib/cli/scripts/resources.ts:22]: I believe either we leave this code or remove it completely.
If you are still testing, then mark the PR as draft, please

**Comment by MichalFidor** [comment]
It will be done in another PR, not in ADF repo
